pkg/operator/ceph/cluster/controller.go

```diff
@@ -439,7 +439,7 @@ func (c *ClusterController) initializeCluster(cluster *cluster, clusterObj *ceph
 
 	if !cluster.Spec.External.Enable {
 		// Start the osd health checker only if running OSDs in the local ceph cluster
-		osdChecker := osd.NewMonitor(c.context, cluster.Namespace)
+		osdChecker := osd.NewMonitor(c.context, cluster.Namespace, cluster.Spec.RemoveOSDsIfOutAndSafeToRemove)
```

Comment:
We need a way to update this inside the osdChecker when the cluster CR changes. Currently, if the cluster CR is updated this will never be updated unless the operator is restarted.